app/src/main/kotlin/co/netguru/baby/monitor/client/feature/communication/pairing/PairingUseCase.kt

```diff
@@ -0,0 +1,110 @@
+import co.netguru.baby.monitor.client.data.client.ChildDataEntity
+import co.netguru.baby.monitor.client.data.client.home.log.LogDataEntity
+import co.netguru.baby.monitor.client.feature.communication.websocket.Message
+import co.netguru.baby.monitor.client.feature.communication.websocket.MessageParser
+import co.netguru.baby.monitor.client.feature.communication.websocket.RxWebSocketClient
+import io.reactivex.disposables.CompositeDisposable
+import io.reactivex.rxkotlin.plusAssign
+import io.reactivex.rxkotlin.subscribeBy
+import io.reactivex.schedulers.Schedulers
+import timber.log.Timber
+import java.net.URI
+import javax.inject.Inject
+
+class PairingUseCase @Inject constructor(
+    private val messageParser: MessageParser,
+    private val rxWebSocketClient: RxWebSocketClient,
+    private val dataRepository: DataRepository,
+    private val localDateTimeProvider: LocalDateTimeProvider
+) {
+    private val compositeDisposable = CompositeDisposable()
+    private val mutablePairingCompletedState = MutableLiveData<Boolean>()
+    internal val pairingCompletedState: LiveData<Boolean> = mutablePairingCompletedState
+
+    fun pair(address: URI, pairingCode: String) {
+        compositeDisposable += rxWebSocketClient.events(address)
+            .subscribeOn(Schedulers.io())
+            .subscribeBy(onNext = {
+                when (it) {
+                    is RxWebSocketClient.Event.Open -> sendPairingCode(pairingCode)
+                    is RxWebSocketClient.Event.Message -> handleMessage(it, address)
+                }
+            }, onError = { mutablePairingCompletedState.postValue(false) })
+    }
+
+    fun cancelPairing() {
+        sendMessage(
+            Message(
+                pairingCode = ""
+            )
+        )
+    }
+
+    fun dispose() {
+        disconnectFromService()
+    }
+
+    private fun sendMessage(message: Message) {
+        compositeDisposable += rxWebSocketClient.send(messageParser.getMessageJson(message))
+            .subscribeBy(
+                onComplete = { Timber.i("message sent: $message") },
+                onError = { Timber.e(it) })
+    }
+
+    private fun sendPairingCode(pairingCode: String) {
+        sendMessage(
+            Message(
+                pairingCode = pairingCode
+            )
+        )
+    }
+
+    private fun disconnectFromService() {
+        rxWebSocketClient.dispose()
+        compositeDisposable.dispose()
+        mutablePairingCompletedState.postValue(false)
+    }
+
+    private fun handleMessage(webSocketMessage: RxWebSocketClient.Event.Message, address: URI) {
+        val message = messageParser.parseWebSocketMessage(webSocketMessage)
+        message?.pairingApproved?.let { pairingApproved ->
+            handlePairingResponse(pairingApproved, address)
+        }
+    }
+
+    private fun handlePairingResponse(pairingApproved: Boolean, address: URI) {
+        if (pairingApproved) handleNewService(address)
```

Comment:
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/netguru/baby-monitor-client-android/blob/9259abfa8c2adcd3ca5c84779794ca9bdaf7e360/app/src/main/kotlin/co/netguru/baby/monitor/client/feature/communication/pairing/PairingUseCase.kt#L82"></span>Multi-line if statement was found that does not have braces. These should be added to improve readability.</td>
    </tr>
  </tbody>
</table>